core/src/main/java/io/grpc/internal/ServerCallImpl.java

```diff
@@ -241,10 +243,17 @@ public void messageRead(final InputStream message) {
         messageReceived = true;
 
         listener.onMessage(call.method.parseRequest(message));
+      } catch (Throwable e) {
+        t = e;
       } finally {
         try {
           message.close();
         } catch (IOException e) {
+          if (t != null) {
+            // TODO(carl-mastrangelo): Maybe log e here.
+            Throwables.propagateIfPossible(t);
```

Comment:
If you are not using propagate() because it's deprecated, so is propagateIfPossible().
We might just always wrap it with RuntimeException. Although it's not ideal to wrap RuntimeException in a RuntimeException, but it doesn't do real harm.